blockchain/chain_sync/src/sync.rs

```diff
@@ -982,6 +971,18 @@ where
     pub fn set_state(&mut self, new_state: SyncState) {
         self.state = new_state
     }
+
+    async fn get_peer(&self) -> PeerId {
+        while self.peer_manager.is_empty().await {
```

Comment:
If this can't find a peer, it will loop indefinitely, is it worth it to have a time out mechanism?

Reply:
Or maybe implement a try_get_peer() that takes in a timeout variable as well?

Reply:
Somewhat intentional, because it can't do anything else until it has a peer to sync with. The logic hasn't changed, I just put in a function because there was another place it should be used. 

The logic is definitely a bit broken now after the logic had been refactored since this was setup, but I don't want to cause conflicts with the work Erlich is doing in a bit. This shouldn't be an issue once the syncing setup is improved, but maybe something to keep in mind @ec2

Reply:
But yeah, definitely agree a timeout is worthwhile to have, just probably after the setup is refactored (unless you guys think otherwise)

Reply:
That is totally fine to me!